src/grisp_pins.erl

```diff
@@ -0,0 +1,64 @@
+-module(grisp_pins).
+
+-export([index/1]).
+-export([map_type/1]).
+-export([attr/1]).
+-export([map_attr/1]).
+-export([bool/1]).
+
+index(gpio1_1)    -> 0;
+index(gpio1_2)    -> 1;
+index(gpio1_3)    -> 2;
+index(gpio1_4)    -> 3;
+index(gpio2_1)    -> 4;
+index(gpio2_2)    -> 5;
+index(gpio2_3)    -> 6;
+index(gpio2_4)    -> 7;
+index(led1_r)     -> 8;
+index(led1_g)     -> 9;
+index(led1_b)     -> 10;
+index(led2_r)     -> 11;
+index(led2_g)     -> 12;
+index(led2_b)     -> 13;
+index(jumper_1)   -> 14;
+index(jumper_2)   -> 15;
+index(jumper_3)   -> 16;
+index(jumper_4)   -> 17;
+index(jumper_5)   -> 18;
+index(spi1_pin9)  -> 19;
+index(spi1_pin10) -> 20;
+index(ss1)        -> 21;
+index(ss2)        -> 22;
+index(Any) when is_integer(Any) ->
+    Any.
+
+%% See: rtems/c/src/lib/libbsp/arm/atsam/libraries/libchip/include/pio.h
+map_type(periph_a) -> 0;
+map_type(periph_b) -> 1;
+map_type(periph_c) -> 2;
+map_type(periph_d) -> 3;
+map_type(input)    -> 4;
+map_type(output_0) -> 5; % default value 0
+map_type(output_1) -> 6. % default value 1
+
+attr(default) -> 0;
+attr(pullup) -> 2#1 bsl 0;
```

Comment:
Instead of the bit shifts we should have a constant here so we don't compute it in runtime every time the mapping is called. For example, instead of `2#1 bsl 6` we should use `2#01000000`. The same goes for the `bor` expressions.

Reply:
I am not sure if users want to register and deregister interrupts often. In many usecases you only register them once during startup in which case I would argue that it is fine doing it like this.

Reply:
I would expect any respectable compiler to execute constant expressions and just do this for us.  Also I agree with @nextl00p that this will be called very rarely.

Reply:
Good point about performance. Don't you think it's clearer with a structure like this though?

```
attr(deglitch)    -> 2#00000010;
attr(opendrain)   -> 2#00000100;
attr(debounce)    -> 2#00001000;
attr(it_aime)     -> 2#00010000;
attr(it_re_or_hl) -> 2#00100000;
attr(it_edge)     -> 2#01000000;
```

If the mapping to the C code is more clear with `bsl`, let's keep it.

Reply:
The original C code can be found here. For me it was the most intuitive way to express it in Erlang.
https://github.com/RTEMS/rtems/blob/master/c/src/lib/libbsp/arm/atsam/libraries/libchip/include/pio.h#L98

Reply:
@nextl00p Since it maps directly to the C code, I think we should keep it.